### DISCUSSION TITLE
Needle vs Factory

### DISCUSSION THREAD

**Discussion by Foixa** [hmlongco/Factory#64] [General]
Can you compare this project with Needle? The Needle seems particularly assertive about performance. How does Factory perform on a large project? Thanks

**Comment by hmlongco** [comment]
Needle is a great project. That said, I never liked the mechanisms it used to define dependencies and I especially never liked adding an additional build step to my project in oder to generate the code. Further, I believe Factory to be somewhat more flexible in how it manages scopes and allows for runtime customization of the dependency cycle.

Needle was one of the benchmarks I used for Factory. Could I have a performant system that didn't require an extra build step, but was also compile-time safe?

Or to put it another way, could I have my cake and eat it too?

Turns out I could.
